tools/automate_aws_os_version.py

```diff
@@ -0,0 +1,118 @@
+import datetime
+import os
+import json
+import re
+import subprocess
+OS_COMMANDS = ['aws ec2 describe-images --owners amazon --filters "Name=name,Values=*Windows_Server-2019-English-Full-Base-*" --query "sort_by(Images, &CreationDate)[].Name"',
+               'aws ec2 describe-images --owners 099720109477 --filters "Name=name,Values=*ubuntu/images/hvm-ssd/ubuntu-bionic-18.04-amd64-server-*" --query "sort_by(Images, &CreationDate)[].Name"',
+               'aws ec2 describe-images --owners aws-marketplace --filters "Name=name,Values=*b7ee8a69-ee97-4a49-9e68-afaee216db2e*" --query "sort_by(Images, &CreationDate)[].Description"']
+PATHS = [["single-connector", "vars.tf"],
+         ["lb-connectors", "vars.tf"]]
+
+
+latest_aws_os = []
+
+def get_os():
+    """
+        get the latest version from AWS
+        """
+    try:
+        for os in OS_COMMANDS:
+            capture_output=subprocess.run(os,capture_output=True).stdout.decode()
+            recent_os=json.loads(capture_output)
+            latest_aws_os.extend([key for key in map(lambda x: x, recent_os)])
+        return latest_aws_os
+    except (AttributeError, TypeError, subprocess.CalledProcessError) as ex:
+        print("Exception occurred: ", ex)
+
+
+
+def compare_versions(latest_os,line):
+    emp=""
+    try:
+        reg=re.compile(r"([a-zA-Z]+)")
+        reg1=re.compile(r"([0-9.])")
+
+        def get_os_name(image_name):
+            match = reg.search(image_name)
+            return match.group(1)
+
+        def get_os_date(image_name):
+            match2=reg1.findall(image_name)
+            listtostr = ' '.join(map(str, match2))
+            os_dates=listtostr.replace(" ","").split("/")
+            for os_date in os_dates:
+                if os_name == "Windows":
+                    return datetime.datetime.strptime(os_date[4:14], "%Y.%m.%d")
```

Comment:
this kind of specific info should be defined per image along with the owner, name, etc.